src/main/java/dev/latvian/mods/kubejs/client/KubeJSKeybinds.java

```diff
@@ -0,0 +1,71 @@
+package dev.latvian.mods.kubejs.client;
+
+import dev.latvian.mods.kubejs.bindings.event.ClientEvents;
+import dev.latvian.mods.kubejs.script.ScriptType;
+import net.minecraft.client.KeyMapping;
+import net.minecraft.client.Minecraft;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class KubeJSKeybinds {
+	private static final Map<String, KubeKeybind> registeredKeybinds = new HashMap<>();
+	private static final List<KubeKeybindState> listeningKeybinds = new ArrayList<>();
```

Comment:
This should be a Set (and KubeKeybindState properly implement equals and hashcode) so that multiple listeners don't result in unnecessary keybind states.